src/conf_mode/protocols_mpls.py

```diff
@@ -121,19 +121,31 @@ def apply(mpls):
         system_interfaces = []
         system_interfaces.append(((os.popen('sysctl net.mpls.conf').read()).split('\n')))
```

Comment:
Please use either `vyos.util.read_file()` or walk the folder with Python hear:

```
>>> import os
>>> from glob import glob
>>> for tmp in glob('/proc/sys/net/mpls/conf/*'):
...   os.path.basename(tmp)
...
'dum0'
'eth0'
'eth0.201'
'eth0.202'
'eth1'
'eth2'
'ip6gre0'
'lo'
```

Reply:
So, while I want to say thank you for this info here as it's very useful...it doesn't give me what I am trying to do.

So the first 4 lines I have give this:

>>> system_interfaces = []
>>> system_interfaces.append(((os.popen('sysctl net.mpls.conf').read()).split('\n')))
>>> del system_interfaces[0][-1]
>>> system_interfaces = system_interfaces[0]
>>>
>>>
>>> print (system_interfaces)
['net.mpls.conf.eth0.input = 0', 'net.mpls.conf.eth0/10.input = 0', 'net.mpls.conf.eth0/11.input = 0', 'net.mpls.conf.eth0/12.input = 0', 'net.mpls.conf.eth0/13.input = 0', 'net.mpls.conf.eth0/14.input = 0', 'net.mpls.conf.eth0/15.input = 0', 'net.mpls.conf.lo.input = 0']


What is super useful here is, it tells you the interface name and the state that it is in. When using glob('/proc/sys/net/mpls/conf/*'), I only get an actual raw dump of interfaces that looks exactly like you have above. When I did it, I got it too:

>>> system_interfaces = []
>>> for interface in glob('/proc/sys/net/mpls/conf/*'):
...     system_interfaces.append(interface.replace('/proc/sys/net/mpls/conf/', ''))
...
>>>
>>> print (system_interfaces)
['eth0', 'eth0.10', 'eth0.11', 'eth0.12', 'eth0.13', 'eth0.14', 'eth0.15', 'lo']

So how should I go about verifying the actual state the interface is in and then making a decision on enabling/disabling?

The reason I ask is because, I would only enable/disable the interfaces that need to have their configuration status changed. I don't want to set all interfaces every time an interface commit happens. So....I am unsure how to proceed.

Reply:
If you wan't the get the current state of all interfaces then please use the `vyos.util.read_file()` wrapper instead of your own `popen()` implementation.